himbaechel/uarch/gatemate/gatemate.cc

```diff
@@ -261,6 +263,45 @@ void GateMateImpl::postPlace()
 {
     repack();
     ctx->assignArchInfo();
+    for (auto &cell : ctx->cells) {
+        // We need to skip CPE_MULT since using CP outputs is mandatory
+        // even if output is actually not connected
+        bool marked_used = cell.second.get()->type == id_CPE_MULT;
+        // Can not use FF for OUT2 if CPE is used in bridge mode
+        if (cell.second.get()->type == id_CPE_FF && ctx->getBelLocation(cell.second.get()->bel).z == CPE_FF_U_Z)
+            marked_used = true;
+        if (marked_used)
+            used_cpes.emplace(ctx->getBelName(cell.second.get()->bel)[0]);
```

Comment:
couldn't this be simplified by just having a pool of tile indices instead and using `bel.tile` in place of `ctx->getBelName(cell.second.get()->bel)[0]` ?